component/newstorage/mem/mem.go

```diff
@@ -92,15 +96,25 @@ func (p *Provider) GetStoreConfig(name string) (newstorage.StoreConfiguration, e
 	return store.config, nil
 }
 
-// Close closes all stores created under this store provider. All data will be deleted.
-func (p *Provider) Close() error {
-	for _, memStore := range p.dbs {
-		err := memStore.Close()
-		if err != nil {
-			return fmt.Errorf("failed to close a mem store: %w", err)
-		}
+// GetOpenStores returns all currently open stores.
+func (p *Provider) GetOpenStores() []newstorage.Store {
+	openStores := make([]newstorage.Store, len(p.dbs))
+
+	var counter int
+
+	for _, db := range p.dbs {
```

Comment:
You can do this:

```
for i, db := range p.dbs{
  openStores[i] = db
}

```

Reply:
`dbs` is a map, not an array, so this won't work.